### DISCUSSION TITLE
Updating an entity that is a discriminant

### DISCUSSION THREAD

**Discussion by ThatOneAwkwardGuy** [mikro-orm/mikro-orm#1976] [Q&A] [answered]
Hi all, I'm currently using Typegoose and thinking about moving to Mikro-orm before I've written too much code in the Typegoose format. Before I move over I have one question (Hopefully it isn't in the docs and I've missed it).

How does one update an entity which is a discriminant?

E.g I have a base class called Asset and multiple things can be an asset, Bank Account, Property, Vehicle .etc
```
class BaseAsset
class BankAccountAsset extends BaseAsset
class PropertyAsset extends BaseAsset
```

If on the client-side, I wish to update an asset by making a put request, can I simply find the asset by ID, wrap it and assign the updated values without specifying what kind of class it is? or can I find the asset as a BaseAsset and then assign the updated values.

```
const asset = await orm.em.findOne(BaseAsset, 1);
wrap(asset).assign({ 
...PropertyAsset information
});
```

Hopefully, this makes sense, happy to elaborate further if needed.

Also has anyone who is using MikoORM found a nice pattern for using the entity classes for DTOs?

**Comment by B4nan** [comment, marked as answer]
Yop, you can query by the base entity, the actual returned instance will be correctly typed based on the discriminator value (TS will see it as the `BaseAsset` but that should not matter much).

**Reply by ThatOneAwkwardGuy** [reply]
Literally amazing, thanks so much for this ORM 😁😁😁
